### PR TITLE
changed the command collection getCommand method to return an error if command not found

### PR DESCRIPTION
**Pull request by Wyvinar**

### CODE REVIEW

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

File: pkg/cmd/propertiesGet_test.go [hunk 1]
```diff
@@ -17,11 +17,12 @@ func TestPropertiesGetCommandInCommandCollectionHasName(t *testing.T) {
 	factory := NewMockFactory()
 	commands, _ := NewCommandCollection(factory)
 
-	propertiesGetCommand := commands.GetCommand(COMMAND_NAME_PROPERTIES_GET)
+	propertiesGetCommand, err := commands.GetCommand(COMMAND_NAME_PROPERTIES_GET)
```
Comment (techcobweb) [inline pkg/cmd/propertiesGet_test.go:20]: if you get an err, you probs need to check that not being nil first ?
Else that's not the error which will show up most prominently when the testcase fail is reported.

File: pkg/cmd/propertiesGet_test.go [hunk 2]
```diff
@@ -17,11 +17,12 @@ func TestPropertiesGetCommandInCommandCollectionHasName(t *testing.T) {
...
 	assert.Equal(t, COMMAND_NAME_PROPERTIES_GET, propertiesGetCommand.Name())
 	assert.NotNil(t, propertiesGetCommand.Values())
 	assert.IsType(t, &PropertiesGetCmdValues{}, propertiesGetCommand.Values())
```
Comment (techcobweb) [inline pkg/cmd/propertiesGet_test.go:23]: Not sure if these asserts should be in an 'if err == nil' block ?

File: pkg/errors/errorMessage.go [hunk 3]
```diff
@@ -206,6 +206,7 @@ var (
 	GALASA_ERROR_DELETE_PROPERTY_RESPONSE_PARSING        = NewMessageType("GAL1117E: The delete operation failed. Unable to process the error information returned from the server.", 1117, STACK_TRACE_WANTED)
 	GALASA_ERROR_RESOURCE_RESPONSE_PARSING               = NewMessageType("GAL1118E: The resource operation failed. Unable to process the error information returned from the server. Reason: '%s'", 1118, STACK_TRACE_WANTED)
 	GALASA_ERROR_RESOURCE_RESP_UNAUTHORIZED_OPERATION    = NewMessageType("GAL1119E: The server thinks you are unauthorized to perform this operation.", 1119, STACK_TRACE_NOT_WANTED)
+	GALASA_ERROR_COMMAND_NOT_FOUND                       = NewMessageType("GAL1120E: '%s' not found in command collection", 1120, STACK_TRACE_NOT_WANTED)
```
Comment (techcobweb) [inline pkg/errors/errorMessage.go:209]: Better to relate this to something the user understands. By all means log a message in the message log saying what's gone wrong,  but the error text should be something simple like "GAL1120E: Program logic error. Collect a log using the '--log' option and send to the Galasa development team."

**Review by techcobweb — APPROVED** [review]
Approved subject to the comments being addressed.

File: pkg/cmd/propertiesGet_test.go (hunk 2 above)
Comment (Wyvinar) [reply pkg/cmd/propertiesGet_test.go:23]: error caught before everything, should exit out before hitting the other asserts now I think?

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-cli-pr-190-57gcs for details. If you are unable to do so, please contact a member of the Galasa team.

**Comment by galasa-team** [comment]
Build successful

**Pull request merged**